src/components/CityDetails/index.jsx

```diff
@@ -0,0 +1,160 @@
+                ),
+            },
+            {
+              text: `Мин.темп: ${Math.trunc(cityData.main.temp_min)}°C`,
+              onClick: () =>
+                handleModalOpen(
+                  `Минимальная температура в данном городе в течении суток: ${Math.trunc(
+                    cityData.main.temp_min
+                  )}°C`
+                ),
+            },
+            {
+              text: `Макс.темп: ${Math.trunc(cityData.main.temp_max)}°C`,
+              onClick: () =>
+                handleModalOpen(
+                  `Максимальная температура в данном городе в течении суток: ${Math.trunc(
+                    cityData.main.temp_max
+                  )}°C`
+                ),
+            },
+            {
+              text: `Температура по ощущениям: ${Math.trunc(
+                cityData.main.feels_like
+              )}°C`,
+              onClick: () =>
+                handleModalOpen(
+                  `«Температура по ощущениям» учитывает поправки на ветер, влажность и даже то, яркое ли сегодня будет солнце. То есть она максимально приближена к тому, что будете чувствовать вы, выйдя на улицу. Температура по ощущениям в данном городе: ${Math.trunc(
+                    cityData.main.feels_like
+                  )}°C`
+                ),
+            },
+            {
+              text: `Видимость: ${cityData.visibility}м`,
+              onClick: () =>
+                handleModalOpen(
+                  `Характеристика прозрачности атмосферы и возможности различать зрением удалённые объекты, отделённые слоем воздуха той или иной мутности. Видимость в данном городе: ${cityData.visibility}м`
+                ),
+            },
+            {
+              text: `Время восхода: ${formattedRise}`,
+              onClick: () =>
+                handleModalOpen(
+                  `Момент появления верхнего края солнечного диска над горизонтом. Время восхода в данном городе: ${formattedRise}`
+                ),
+            },
+            {
+              text: `Время заката: ${formattedSunset}`,
+              onClick: () =>
+                handleModalOpen(
+                  `Время захода солнца определяют в астрономии как момент, когда верхний край солнечного диска исчезает за горизонтом. Время заката в данном городе: ${formattedSunset}`
+                ),
+            },
+          ].map((item, index) => (
+            <a key={index} onClick={item.onClick}>
+              {item.text}
+            </a>
+          ))}
+        </div>
+
+        <div>
+          <Link className="details_back" to="javascript:history.back()">
+            Назад
+          </Link>
+        </div>
+      </div>
+      {isModalOpen && (
+        <Modal
+          content={modalContent}
+          onClose={handleModalClose}
+          isModalOpen={isModalOpen}
+        />
+      )}
```

Comment:
I am not sure I understand why there 2 duplicated conditions here: `isModalOpen &&` and `isModalOpen={isModalOpen}`. AFAIK, if we pass `isModalOpen={false}`, the modal should disappears. 

```suggestion
      </div>
      
      {isModalOpen && (
        <Modal
          content={modalContent}
          onClose={handleModalClose}
          isModalOpen={isModalOpen}
        />
      )}
```

Reply:
I checked, if you pass false, it closes immediately when you open it, so you need to pass true

Reply:
So, do we need `{isModalOpen && (` here?